pointslicepool/pointslicepool.go

```diff
@@ -51,9 +51,9 @@ func (p *PointSlicePool) PutMaybeNil(s []schema.Point) {
 }
 
 func (p *PointSlicePool) Put(s []schema.Point) {
-	if cap(s) >= p.defaultSize {
+	if cap(s) > p.defaultSize {
 		p.putLarge.Inc()
-	} else {
+	} else if cap(s) < p.defaultSize {
```

Comment:
So we no longer track puts and get-makes if they equal defaultSize.
I can see how that helps with troubleshooting the kind of efficiencies you're looking into. OTOH only having metrics for a subset of the puts and get-makes is a bit weird, especially because the sum no longer relates to the get-candidate metrics. should we introduce get-make-default and put-default (or get-make-min and put-min) to track these? or are all these metrics getting out of hand (seems we rarely use them. but then again, that goes for most of the metrics i guess)

Reply:
Yeah, I debated this back and forth with myself.  I don't want to over-instrument the code and these metrics aren't generally interesting except when trying to determine the efficacy of the slice pool in reducing allocations (which is exactly what I was just doing).

Ultimately, it would be nice to know:

- Allocations/bytes saved using the pool
- Allocation/bytes required through the pool (and maybe *additional* bytes due to default size)
- Reason the allocation is required (miss or unfit)

I'm not sure I find the "put" metrics that interesting, except maybe in determining that we are "losing" slices that should be returned.

Reply:
I lean towards adding in get-make-min and put-min, to make the set of metrics coherent.
if we decide the metrics are not (or no longer) useful, we can take them out all at once.